SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -87,55 +135,77 @@ TArray<SpatialLoadBalanceEnforcer::AclWriteAuthorityRequest> SpatialLoadBalanceE
 	TArray<Worker_EntityId> CompletedRequests;
 	CompletedRequests.Reserve(AclWriteAuthAssignmentRequests.Num());
 
-	for (WriteAuthAssignmentRequest& Request : AclWriteAuthAssignmentRequests)
+	for (Worker_EntityId EntityId : AclWriteAuthAssignmentRequests)
 	{
-		const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
+		const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(EntityId);
 		if (AuthorityIntentComponent == nullptr)
 		{
-			// TODO(zoning): Not sure whether this should be possible or not. Remove if we don't see the warning again.
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("(%s) Entity without AuthIntent component will not be processed. EntityId: %lld"), *WorkerId, Request.EntityId);
-			CompletedRequests.Add(Request.EntityId);
+			// This happens if the authority intent component is removed in the same tick as a request is queued.
```

Comment:
Is this something that's expected behaviour then? If so, the message should be a Log not a Warning.

Reply:
This flow produced a warning before, in process acl assignments. Kept it as a warning here for consistency. Can change if you would rather it was a log.

Reply:
Actually, now that you remove the queued request when the component is removed, this shouldn't ever happen, right? In which case it should be an error.

Reply:
changed to an error.